RecoTauTag/RecoTau/src/AntiElectronIDMVA6.cc

```diff
@@ -412,17 +485,345 @@ double AntiElectronIDMVA6<TauType, ElectronType>::MVAValue(const TauVars& tauVar
   }
   return mvaValue;
 }
+////
+template <class TauType, class ElectronType>
+double AntiElectronIDMVA6<TauType, ElectronType>::MVAValuePhase2(const TauVars& tauVars,
+                                                                 const TauGammaMoms& tauGammaMoms,
+                                                                 const ElecVars& elecVars) {
+  if (!isInitialized_) {
+    throw cms::Exception("ClassNotInitialized") << " AntiElectronMVA6 not properly initialized !!\n";
+  }
+
+  double mvaValue = -99.;
 
+  //do not consider tau candidates outside the HGCal border at |eta|=3
+  if (std::abs(tauVars.etaAtEcalEntrance) > 3.0) {
+    return mvaValue;
+  }
+
+  float elecDeltaPinPoutOverPin = (elecVars.pIn > 0.0) ? (std::abs(elecVars.pIn - elecVars.pOut) / elecVars.pIn) : 1.0;
+  float elecEecalOverPout = (elecVars.pOut > 0.0) ? (elecVars.eEcal / elecVars.pOut) : 20.0;
+  float elecNumHitsDiffOverSum =
+      ((elecVars.gsfNumHits + elecVars.kfNumHits) > 0.0)
+          ? ((elecVars.gsfNumHits - elecVars.kfNumHits) / (elecVars.gsfNumHits + elecVars.kfNumHits))
+          : 1.0;
+
+  if (tauGammaMoms.signalPFGammaCandsIn == 0 && tauVars.hasGsf < 0.5) {
+    if (std::abs(tauVars.etaAtEcalEntrance) < ecalBarrelEndcapEtaBorder_) {
+      Var_NoEleMatch_woGwoGSF_Barrel_[0] = std::min(2.f, tauVars.leadChargedPFCandPt / std::max(1.f, tauVars.pt));
+      Var_NoEleMatch_woGwoGSF_Barrel_[1] = std::log(std::max(1.f, tauVars.pt));
```

Comment:
some pointers to the variable alignment document may be useful.
I see that in many cases the set of variables is repeated/identical except for an occasional insert of another variable to make the rest appear out of order.
With more appropriate ordering the following about 300 lines could be coded in well under 100 with a smaller chance of mistakes.

I would still suggest to group cases with more than 5 repeated values appearing in the same order into functions and use them

Reply:
I tend to agree that this piece of code can be refactored, but I prefer to not change it. There are several tables with not exactly same content, order and length setting up different BDTs.